PD-285   Write tutorial for Google Photos credentials created with Google API (#1871)
truenas/documentation: content/SCALE/SCALETutorials/DataProtection/CloudSyncTasks/AddGooglePhotosCloudSyncTask.md

---
title: "Adding a Google Photos Cloud Sync Task"
description: "This article provides instructions on how to set up Google Photos API credentials and use them to create a cloud sync task."
weight: 40
tags:
- scalecloud
- scalesgooglephotos
- scalebackup
---

{{< toc >}}

Google Photos works best in TrueNAS using a Google Photos API key and [rclone](https://rclone.org/) token.

## Creating the API Credentials

> On the [Google API dashboard](https://console.cloud.google.com/apis/dashboard), click the dropdown menu next to the Google Cloud logo and select your project. If you do not have a project, click **NEW PROJECT** and enter a **Project name**, **Organization**, and **Location**.

Reviewer: small change suggestion. As a style suggestion (from most of the style guides out there), we don't want to use the UI elements in this way so I suggest this:
```suggestion
On the [Google API dashboard](https://console.cloud.google.com/apis/dashboard), click the dropdown menu next to the Google Cloud logo and select your project. If you do not have a project, click **NEW PROJECT** and enter a value in **Project name**, **Organization**, and **Location**.
```

![GooglePhotosAPINewProject](/images/SCALE/22.12/GooglePhotosAPINewProject.png "New Project")

### Enable API

After you select your project, click **Enabled APIs & Services** on the left menu, then click **+ ENABLE APIS AND SERVICES**.

![GooglePhotosAPIEnableAPIsandServices](/images/SCALE/22.12/GooglePhotosAPIEnableAPIsandServices.png "Enable APIs and Services")

> Enter *photos library api* in the search bar, then select **Photos Library API** and click **ENABLE**.

Reviewer: is this a variable or what they actually enter in the search field? I know it is a lot of bold, but if it is specific we use bold for those values. If this is just one way to search for it, italics is okay but clarify this is a way to search for it.
```suggestion
Enter *photos library api* in the search bar, then select **Photos Library API** and click **ENABLE**.
```

![GooglePhotosAPIPhotosLibraryAPISearch](/images/SCALE/22.12/GooglePhotosAPIPhotosLibraryAPISearch.png "Photos Library API Search")

![GooglePhotosAPIClickEnable](/images/SCALE/22.12/GooglePhotosAPIClickEnable.png "Click Enable")

### Configure Authentication

Next, click **OAuth consent screen** on the left menu, select **EXTERNAL**, then click **CREATE**.

![GooglePhotosAPICreateExternal](/images/SCALE/22.12/GooglePhotosAPICreateExternal.png "Create External")

> Enter an **App name** and **User support email**. 

Reviewer: Same reason as above. It is best to not use the field name as the "value" to enter.
```suggestion
Enter a value in **App name** and **User support email**. 
```

![GooglePhotosAPIAppInformation](/images/SCALE/22.12/GooglePhotosAPIAppInformation.png "Enter App Information")

Enter an email address in the **Developer contact information** section, then click **SAVE AND CONTINUE**.

![GooglePhotosAPIDeveloperContactInformation](/images/SCALE/22.12/GooglePhotosAPIDeveloperContactInformation.png "Enter Developer Contact Information")

Continue to the **Add users** section, enter your email address, then click **ADD**.

![GooglePhotosAPIAddTestingUser](/images/SCALE/22.12/GooglePhotosAPIAddTestingUser.png "Add Testing User")

On the **OAuth consent screen**, click **PUBLISH APP** under **Testing** and push the app to production.

![GooglePhotosAPIPublish](/images/SCALE/22.12/GooglePhotosAPIPublish.png "Publish Status")

{{< expand "Can I leave the app in testing mode?" "v" >}}

> You can leave the app in testing mode, but your cloud sync task will fail after your testing app credentials expire after seven days. 

Reviewer: active voice, present tense suggestion
```suggestion
You can leave the app in testing mode, but your cloud sync task fails after your testing app credentials expire after seven days. 
```

![GooglePhotosAPICreateOAuthClientID](/images/SCALE/22.12/GooglePhotosAPICreateOAuthClientID.png "Create OAuth Client ID")

Copy and save your client ID and secret, or download the JSON file.

![GooglePhotosAPICopyIDandSecret](/images/SCALE/22.12/GooglePhotosAPICopyIDandSecret.png "Copy ID and Secret")

## Configuring Rclone

Download [rclone](https://rclone.org/downloads/) for your OS and open it in a command line utility. The example photos in this article use Powershell in Windows OS.

Enter `rclone config`, then enter `n` to create a new remote.

![GooglePhotosAPIrcloneConfig](/images/SCALE/22.12/GooglePhotosAPIrcloneConfig.png "Configure rclone")

Enter a name for the new remote, then enter the number from the list corresponding to Google Photos.

Enter the client id and secret you saved when you created the Google Photos API credentials, then enter `false` to keep the Google Photos backend read-only.

![GooglePhotosAPIrcloneConfig2](/images/SCALE/22.12/GooglePhotosAPIrcloneConfig2.png "Configure rclone")

Do not edit the advanced config. When prompted about automatically authenticating rclone with the remote, enter `y`.

> A browser window opens to authorize rclone access. Click **Allow*.

Reviewer: ```suggestion
A browser window opens to authorize rclone access. Click **Allow**.
```

In the command line, enter `y` when prompted about media item resolution to complete the configuration.

> Copy and save the Options (type, client_id, client_secret, token), then enter `y` to keep the new remote.

Reviewer: is Options something on the screen? if yes use bold, if not or a general reference to options use lower case

> Copy and save the Options (type, client_id, client_secret, token), then enter `y` to keep the new remote.

Author: ```suggestion
Copy and save the type, client_id, client_secret, and token, then enter `y` to keep the new remote.
```
